include/aos/common/tools/error.hpp

```diff
@@ -99,11 +99,27 @@ class Error {
             mLineNumber = err.mLineNumber;
         }
 
+        if (msg && err.mMessage[0] != '\0') {
```

Comment:
We should not concatenate error.  The logic should be as following if `msg` is not `nullptr` and `mMessage` is empty than set `msg` to `mMessage`. Do nothing, otherwise.